components/apimgt/org.wso2.carbon.apimgt.gateway.extension/src/main/java/org/wso2/carbon/apimgt/gateway/internal/ConfigurationActivator.java

```diff
@@ -43,15 +43,17 @@
             unbind = "unregisterConfigProvider")
     protected void registerConfigProvider(ConfigProvider configProvider) {
         ServiceReferenceHolder.getInstance().setConfigProvider(configProvider);
-
-        /*try {
-           AnalyticsConfiguration analyticsConfiguration = ServiceReferenceHolder.getInstance().getConfigProvider()
+        AnalyticsConfiguration analyticsConfiguration = null;
+        try {
+            analyticsConfiguration = ServiceReferenceHolder.getInstance().getConfigProvider()
                     .getConfigurationObject(AnalyticsConfiguration.class);
         } catch (CarbonConfigurationException e) {
-            log.error("error getting config", e);
-        }*/
-
-        AnalyticsConfiguration analyticsConfiguration = new AnalyticsConfiguration();
+            log.error("error getting config : AnalyticsConfiguration", e);
+        }
+        if (analyticsConfiguration == null) {
+            analyticsConfiguration = new AnalyticsConfiguration();
+            log.info("org.wso2.carbon.apimgt.gateway.interna.ConfigurationActivator : Setting default configurations");
```

Comment:
This has fixed in recent commit 8d91f6afbed8052502a3e2c44b6e366cc288bf23
Thanks.